libp2p/connmanager.nim

```diff
@@ -105,15 +109,29 @@ proc addConnEventHandler*(c: ConnManager,
   ## Add peer event handler - handlers must not raise exceptions!
   ##
 
-  if isNil(handler): return
-  c.connEvents.mgetOrPut(kind,
-    initOrderedSet[ConnEventHandler]()).incl(handler)
+  try:
+    if isNil(handler): return
+    c.connEvents.mgetOrPut(kind,
+      initOrderedSet[ConnEventHandler]()).incl(handler)
+  except Exception as exc:
```

Comment:
with https://github.com/status-im/nim-libp2p/pull/573, these should no longer be needed :) That said, once `ConnEventHandler` is marked as `raises: [Defect]` the bug goes away :/

Reply:
weird, I'm still getting the `Exception` with Nim 1.2.6.